controllers/util/mw_util.go

```diff
@@ -524,7 +530,17 @@ func (mwu *MWUtil) DeleteManifestWorksForCluster(clusterName string) error {
 		return fmt.Errorf("failed to delete ManifestWork for VRG in namespace %s (%w)", clusterName, err)
 	}
 
-	// The ManifestWork that created a Namespace is intentionally left on the server
+	return nil
+}
+
+func (mwu *MWUtil) DeleteManifestWorksForClusterNamespace(clusterName string) error {
```

Comment:
A better name would be DeleteNamespaceManifestWorksForCluster and the older function should be renamed to DeleteVRGManifestWorksForCluster.

Or add the code to delete the ns mnifestwork in the older function instead of the comment. I think the second option is better, why complicate the callers when we can handle it in the MWUtil helper?

Reply:
This is because we want to delete the namespace manifest work only when delete function is called from finalizeDRPC() and not when it is called from ensureVRGManifestWorkOnClusterDeleted()

Reply:
Good point, but this intention is not communicated anywhere. The commit message should describe the problem you are trying to solve, and how the problem is solved.

Reply:
Yeah sure, I will include the description with final changes.